cpu_scheduling_algorithms/round_robin_scheduling.cpp

```diff
@@ -0,0 +1,242 @@
+/**
+ * @file
+ * @brief Implementation of Round Robin CPU scheduling algorithm
+ * @details
+ * Round-robin is a preemptive CPU scheduling algorithm where each
+ * ready task runs turn by turn only in a cyclic queue for a limited
+ * time slice. This algorithm also offers starvation free execution
+ * of processes.
+ * @author [Daemon19](https://github.com/Daemon19)
+ */
+
+#include <iomanip>  // For formatting process results output
+#include <cassert>  // For testing the round-robin algorithm
+#include <string>   // For converting int type to string
+#include <iostream> // For outputting process execution results
+#include <queue>    // Container for process execution turn
+#include <set>      // Container for processes that have arrived
+#include <utility>  // So I can use std::pair
+#include <vector>   // Container for processes that will be executed
+
+/**
+ * @brief Represent a process to be executed.
+ */
+struct Process {
+    uint32_t id;
+    uint32_t arrival_time;
+    uint32_t burst_time;
+};
+
+/**
+ * @brief Represent the result of a process execution.
+ */
+struct ProcessResult : public Process {
+    uint32_t completion_time;
+    uint32_t turn_around_time;
+    uint32_t waiting_time;
```

Comment:
Please add a one-line description of what the variable is for (**see the example below**).

```c++
int64_t data = 0;       ///< The value/key of the node.
```